classes/01class/exercises/c01-git04/igorjosesantos/README.md

```diff
@@ -0,0 +1,50 @@
+# c01-git04
+
+## Questions (answer inline)
+
+1. Let's suppose you remotely pushed the `my_env.txt` file above. A colleague asks you to remove this information from Git. What's your colleague worried about?
+
+> He's worried about the sensitive data information present in that file. In that case, credentials.
+
+2. If you modify the file in your workspace, then commit and push it, will it be enough to erase this password information from the repository? (It's not). Why?
+
+> No. The sensitive information still present in the history and could be retrieved from there.
+
+3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?
+
+> No. It's a different approach with the same result as above. The data still in the history.
+
+4. How to fix this? How do you remove something from Git history when it is in the remote repository?
+
+> It's necessary to rewrite the history and (force) push to remove the entries from the remote history.
```

Comment:
What do you think are the security complications of that? How safe do you think the credentials are after you force-pushed and removed it from the remote history?

Reply:
You have to change your credentials anyway. Unless you're sure that no one saw that.

Reply:
If public repo we have to consider it compromised, there are bots scanning GitHub and gitlab looking for passwords and private keys...